kie-server-parent/kie-server-services/kie-server-services-drools/src/main/java/org/kie/server/services/drools/DroolsKieSessionLookupHandler.java

```diff
@@ -65,7 +65,9 @@ public void postLookupKieSession(String kieSessionId, KieContainerInstance conta
             //default handler
             PrometheusMetricsDroolsListener listener = new PrometheusMetricsDroolsListener(PrometheusKieServerExtension.getMetrics(),
                     kieSessionId, containerInstance);
-            eventManager.addEventListener(listener);
+            if (!eventManager.getAgendaEventListeners().contains(listener)) {
```

Comment:
Sorry, I know I'm being fussy here, but I'm not sure if this is the best way to fix the problem. In particular I don't think that a listener should implement equals/hashCode and moreover doing so in most cases you're just creating a listener instance with the only purpose to compare it and then throw it away.

I assume that it makes sense to only have one `PrometheusMetricsDroolsListener` for each and every session and if that's true I'd prefer to replace this fix with something like the following
```

if (eventManager.getAgendaEventListeners().stream().noneMatch(PrometheusMetricsDroolsListener.class::isInstance)) {
    PrometheusMetricsDroolsListener listener = new PrometheusMetricsDroolsListener(PrometheusKieServerExtension.getMetrics(),
            kieSessionId, containerInstance);
    eventManager.addEventListener(listener);
}
```

Reply:
@mariofusco Thank you! You are absolutely right. Fixed.